recordCollection.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/miekg/dns"
+)
+
+type (
+	recordCollection []cloudflare.DNSRecord
+)
+
+// Find will serach for needle in a recordCollection.
+func (c recordCollection) Find(needle cloudflare.DNSRecord) (int, *cloudflare.DNSRecord) {
+	for i, r := range c {
+		if match(r, needle) {
+			return i, &r
+		}
+	}
+
+	return -1, nil
+}
+
+// Diff will find the differences between two recordCollections.
+func (c recordCollection) Diff(remote recordCollection) (recordCollection, recordCollection) {
+	localOnly := recordCollection{}
+	remoteOnly := recordCollection{}
+
+	for _, l := range c {
+		n, _ := remote.Find(l)
+
+		if n < 0 {
+			localOnly = append(localOnly, l)
+		}
+	}
+
+	for _, r := range remote {
+		n, _ := c.Find(r)
+
+		if n < 0 {
+			remoteOnly = append(remoteOnly, r)
+		}
+	}
+
+	return localOnly, remoteOnly
+}
+
+func (c recordCollection) Print(w io.Writer) {
```

Comment:
Maybe "WriteZoneFile" is a better name for this. It might as least be worth a comment that this writes in zone file format.

Reply:
Added a comment and renamed to Fprintf() in ed577e3.